tests/proxy/test_argument_handling.py

```diff
@@ -12,3 +15,58 @@ def test_bad_push_items():
 
     with pytest.raises(jsonschema.ValidationError):
         coll.update_push_items([{"foo": "bar"}])
+
+
+def test_pushitem_obj_push():
+    """PushItem object passed to the update_push_items works fine."""
+
+    collector = Collector.get("dummy")
+
+    pushitem = PushItem(name="test_pushitem")
+    ret_val = collector.update_push_items([pushitem])
+
+    assert ret_val.result() is None
+
+
+def test_bad_obj_push():
+    """Any object apart from PushItem type raises a validation error."""
+
+    collector = Collector.get("dummy")
+
+    pushitem = object()
+    with pytest.raises(jsonschema.ValidationError):
+        collector.update_push_items([pushitem])
+
+
+def test_pushitem_obj_attributes():
+    """PushItem object attributes are translated and available as expected in
+    pushitem dict passed to update_push_items"""
+
+    Collector.register_backend("mock", Mock)
+    collector = Collector.get("mock")
+
+    pushitem = PushItem(
+        name="test_push",
+        origin="some_origin",
+        src="source",
+        dest=["dest1", "dest2"],
+        md5sum="bb1b0d528129f47798006e73307ba7a7",
+        sha256sum="4fd23ae44f3366f12f769f82398e96dce72adab8e45dea4d721ddf43fdce31e2",
+        build="test_build",
+        signing_key="FD431D51",
+    )
+    collector.update_push_items([pushitem])
+
+    update_push_item_args = collector._delegate.update_push_items.call_args[0][0]
```

Comment:
This test relies on the implementation detail that get() returns an object with some _delegate pointing at the backend object.  This is unrelated to what you want to test.  Can we please avoid unnecessary dependencies on details like that?  It increases the cost of refactoring.

I don't see any reason the test needs to be written in this way.  Your test right now sets up the backend to create a new Mock object whenever an instance is requested, and you then need to rely on some implementation details to get a reference to that object.  Why not just register your backend to return a mock for which you already hold a reference:

```
mock = Mock()
# This backend always returns the mock we prepared.
Collector.register_backend("mock", lambda: mock)
collector = Collector.get("mock")

...

update_push_item_args = mock.update_push_items...
```

Reply:
+1